subsys/bluetooth/scan.c

```diff
@@ -295,11 +295,15 @@ static void scan_connect_with_target(struct bt_scan_control *control,
 	bt_scan_stop();
 
 	/* Establish connection. */
-	struct bt_conn *conn = bt_conn_create_le(addr, &bt_scan.conn_param);
+	struct bt_conn *conn;
+
+	rc = bt_conn_le_create(device_info->addr,
+			       BT_CONN_LE_CREATE_CONN,
+			       device_info->conn_param, &conn);
 
 	LOG_DBG("Connecting");
 
-	if (!conn) {
+	if (rc || !conn) {
```

Comment:
Checking `rc` should be enough, also I'd prefer `err` which is consistent with the rest of the file.

Reply:
done